src/Microsoft.OData.Core/Json/BufferingJsonReader.cs

```diff
@@ -338,7 +338,7 @@ public virtual async Task<Stream> CreateReadStreamAsync()
 
             if (value == null)
             {
-                result = new MemoryStream(new byte[0]);
+                result = Stream.Null;

```

Comment:
Is there a difference between using 
``` 
result = null
```
and 

```
result = Stream.Null

```

Reply:
@ElizabethOkerio Yes. `result = null` is equivalent of `string[] array = null` while `result = Stream.Null` is the equivalent of `string[] array = Array.Empty<string>();`

`Stream.Null` is a stream with no backing store, which is not the same as null.

https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.null?view=net-6.0

Reply:
We're doing this to avoid unnecessary allocations. When you do `new byte[0]`, an allocation is done. Think of `Stream.Null` as a **reusable** zero-length array that helps avoid repeated zero-length allocations